src/foam/nanos/ruler/RuledDAO.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.ruler',
+  name: 'RuledDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'Finds applicable ruled (i.e, rule-like) object by FindRuledCommand.',
+
+  javaImports: [
+    'foam.dao.ArraySink',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'cmd_',
+      javaCode: `
+        if ( obj instanceof FindRuledCommand ) {
+          var sink = (ArraySink) getDelegate()
+            .where(AND(
+              INSTANCE_OF(Ruled.getOwnClassInfo()),
+              EQ(Ruled.RULE_GROUP, ((FindRuledCommand) obj).getRuleGroup()) ))
+            .orderBy(DESC(Ruled.PRIORITY))
+            .select(new ArraySink());
```

Comment:
Better to process directly in the sink, then, you can stop once you find the first match.

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/4484/commits/d81b5809a06ff4d4eb5a0b2527a3f81bf6314cce.